kornia/feature/nms.py

```diff
@@ -30,10 +48,16 @@ def pad(x):
 
     def forward(self, x: torch.Tensor, mask_only: bool = False) -> torch.Tensor:  # type: ignore
         assert len(x.shape) == 4, x.shape
+        B, CH, H, W = x.size()
         # find local maximum values
+        max_non_center = F.conv2d(x, self.kernel.repeat(CH, 1, 1, 1).to(x.device, x.dtype),
+                                  stride=1,
+                                  padding=self.padding,
+                                  groups=CH).view(B, CH, -1, H, W).max(dim=2)[0]
```

Comment:
wouldn't adaptive_maxpool2d be more suitable for that ? do you know performance max vs adaptive_max_pool?

Reply:
https://arxiv.org/pdf/2003.13630.pdf Those guys did opposite :)

"We found that a simple dedicated implementation of
GAP, using PyTorch View and Mean tensor operations, can
be up to 5 times faster then AvgPool2d on GPU. Another
optimization we offer is flattening the average pooling in the
final layer, thus avoiding additional reshaping. Our TResNet implementation for fixed GAP appears in appendix A."

Reply:
great! good to know then